src/main/java/io/codelex/luminor/service/PaymentService.java

```diff
@@ -0,0 +1,91 @@
+package io.codelex.luminor.service;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.codelex.luminor.model.Payment;
+import io.codelex.luminor.repository.PaymentRepository;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+@Service
+public class PaymentService {
+
+    private final PaymentRepository paymentRepository;
+    private final RestTemplate restTemplate = new RestTemplate();
+
+    @Value("${geoapify.api.key}")
+    private String apiKey;
+
+    public PaymentService(PaymentRepository paymentRepository) {
+        this.paymentRepository = paymentRepository;
+    }
+
+    public String resolveCountryCode(String ip) {
```

Comment:
Valsts izgūšana liekas, ka varētu būt drīzāk citā servisā - īsti ar maksājumiem tas saistīts nav. Ja Tavā aplikācijā nāktu klāt kaut kas ar maksājumiem nesaistīts, kur arī jaizgūst IP lokācija, tad nebūtu jēdzīgi, ka šis ir iekš payment service.